packages/@react-stately/data/src/useListData.ts

```diff
@@ -232,10 +236,11 @@ export function createListActions<T>(opts: ListOptions<T>, dispatch: (updater: (
     removeSelectedItems() {
       dispatch(state => {
         if (state.selectedKeys === 'all') {
+          // return 'all' value if there still exists unloaded items (i.e. cursor is not null)
           return {
             ...state,
             items: [],
-            selectedKeys: new Set()
+            selectedKeys: cursor != null ? 'all' : new Set()
```

Comment:
Wouldn't the previous behavior make sense here though? If `all` was previously selected, and those items were removed, then that means that all items, even those not yet loaded have been deleted.